bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,417 @@
+                        if sub_sub_item["kind"] == "NAMESPACE_REF":
+                            namespace_ref += f'{sub_sub_item["name"]}::'
+                        if sub_sub_item["kind"] == "TYPE_REF":
+                            parameter_type_list.append(
+                                f'{namespace_ref}{sub_sub_item["name"]}'
+                            )
+                elif sub_item["element_type"] in ("Float", "Int"):
+                    parameter_type_list.append(f'{sub_item["element_type"].lower()}')
+                else:
+                    parameter_type_list.append(f'{sub_item["element_type"]}')
+        parameter_type_list = ",".join(parameter_type_list)
 
-def handle_namespace(item):
-    namespace = item["identifier"]
-    module_linelist.append(f"namespace {namespace}" + "{")
-    for sub_item in item["members"]:
-        type_functions[sub_item["type"]](sub_item)
-    module_linelist.append("}")
+        # default ctor `.def(py::init<>())` already inserted while handling struct/class decl
+        if parameter_type_list:
+            self._linelist.append(f".def(py::init<{parameter_type_list}>())")
 
+    # TODO: Remove, maybe
+    def handle_class_template(self):
+        flag = False
 
-def handle_struct(item):
-    global in_struct
-    in_struct = True
-    name_decl = item["identifier"]
-    name_impl = item["identifier"]
-    if "parent" in item.keys():
-        name_impl = f'{item["identifier"]}, {item["parent"]}'
+        # TODO: Use list based method, like in handle_struct_decl
+        for sub_item in self.members:
+            if sub_item["kind"] == "TEMPLATE_NON_TYPE_PARAMETER":
+                self._linelist.append(
+                    f'template< {sub_item["element_type"].lower()} {sub_item["name"]} >'
+                )
+                flag = True
+        if not flag:
+            self._linelist.append(f"template<>")
+        cxx_base_specifier_list = [
+            sub_item["name"]
+            for sub_item in self.members
+            if sub_item["kind"] == "CXX_BASE_SPECIFIER"
+        ]
+        if cxx_base_specifier_list:
+            cxx_base_specifier_list = ",".join(cxx_base_specifier_list)
+            self._linelist.append(
+                f'py::class_<{self.name, cxx_base_specifier_list}>(m, "{self.name}")'
+            )
+        else:
+            self._linelist.append(f'py::class_<{self.name}>(m, "{self.name}")')
+
+    def handle_inclusion_directive(self):
+        """
+        Handle `CursorKind.INCLUSION_DIRECTIVE`
+        """
+
+        # TODO: develop
+        pass
 
-    module_linelist.append(f'py::class_<{name_impl}>(m, "{name_decl}")')
-    for sub_item in item["members"]:
-        type_functions[sub_item["type"]](sub_item)
-    module_linelist.append(";")
-    in_struct = False
+        # TODO: update blacklist
+        # blacklist = ["pcl/memory.h", "pcl/pcl_macros.h",]
+        # if self.name.startswith("pcl"):
+        #     self._inclusion_list.append(self.name)
```

Comment:
Is it possible to factor out `"pcl"` to the class ctor?

_Originally posted by @kunaltyagi in https://github.com/divmadan/pcl/pull/8_

Reply:
Done.